src/main/java/spoon/support/compiler/SnippetCompilationHelper.java

```diff
@@ -50,13 +50,12 @@ private SnippetCompilationHelper() { }
 
 	public static void compileAndReplaceSnippetsIn(CtType<?> c) {
 		Factory f = c.getFactory();
-		CtType<?> workCopy = c;
 		Set<ModifierKind> backup = EnumSet.noneOf(ModifierKind.class);
-		backup.addAll(workCopy.getModifiers());
-		workCopy.removeModifier(ModifierKind.PUBLIC);
+		backup.addAll(c.getModifiers());
```

Comment:
Ok with the remove of workCopy but then could you rename the argument of the method: `c` is really not something explicit... `workType` maybe or whatever.